clesto/__init__.py

```diff
@@ -806,9 +806,201 @@ def __init__(self, data=None, torsion=None):
         # initialize element
         super(Surjection_element, self).__init__(data=data, torsion=torsion)
 
+    def pcompose(self, other, k):
+        '''partial composition'''
+
+        class Piece:
+            def __init__(self, piece, is_in_surj1, index_in_surj):
+                self.piece = piece
+                self.is_in_surj1 = is_in_surj1
+                self.index_in_surj = index_in_surj
+
+            def __getitem__(self, index):
+                return self.piece[index]
+
+            def __len__(self):
+                return len(self.piece)
+
+            def __eq__(self, other):
+                if isinstance(other, Piece):
+                    result = self.piece == other.piece
+                    result &= self.is_in_surj1 == other.is_in_surj1
+                    result &= self.index_in_surj == other.index_in_surj
+                    return result
+                return False
+
+        def table_arrangement(surj, only_dict=False):
+            '''Returns the table arrangement of a surjection, as a tuple.
+               If only_dict=True, it returns a dictionary dict such that
+               dict[index] is the table arrangement of surj where
+               surj[index] lies.
+
+               '''
+            finals = set()
+            for k in range(1, max(surj)+1):
+                for index in reversed(range(len(surj))):
+                    if surj[index] == k:
+                        finals.add(index)
+                        break
+
+            if only_dict:
+                result = dict()
+                row = 0
+                for index, el in enumerate(surj):
+                    result[index] = row
+                    if index not in finals or index == len(surj)-1:
```

Comment:
I think it would be good to talk about our policy regarding operator/whitespaces. So far I am following the first 3 of
E225 - missing whitespace around operator.
E226 - missing whitespace around arithmetic operator.
E227 - missing whitespace around bitwise or shift operator.
E228 - missing whitespace around modulo operator.